staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
```

Comment:
@liggitt Why you think it's odd, `clientcmdapi.AuthProviderConfig` is the config load from kubeconfig, and the `BearerTokenSource` is used to generate a token, like the `WrapTransport` func in the config struct, and currently the AuthProvider are generate both including the `WrapTransport` and `AuthProviderConfig`.